test/fixed-queue.ts

```diff
@@ -65,7 +65,7 @@ test('remove not queued task should not lead to errors', async ({ equal }) => {
 test('removing elements from intermediate CircularBuffer should not lead to issues', async ({ equal, same }) => {
   const queue = new FixedQueue();
 
-  const batchSize = 2048;
+  const batchSize = 2047;
```

Comment:
Does it fails with 2048?
In the worst case, it will pre-allocate another buffer, isn't it?

Reply:
In general it doesn't fail but it should match the batch size due to test's intensions.  
The test may require additional clarification with comments.  
The main goal was to check following scenario:
1. We fill the queue with 3 full circular buffers amount of items.
2. Empty the middle circular buffer.
3. This should lead to the removal of the middle buffer from the queue:
  - Before emptying: tail buffer -> middle buffer -> head buffer.
  - After emptying: tail buffer -> head buffer.

My initial `remove` implementation didn't remove empty buffer so I wrote a test to fix it.

Reply:
Nice, now is clear; maybe add that to the description of the test can help understand why are we doing that

Reply:
Although I added the description I must admit that the test is actually insufficient/fragile due to lack of internal checks.  
As you rightly noticed it doesn't and it wouldn't break even if the size of internal circular buffer will ever changes.  
We potentialy could try to really check the amount of buffers with something like following, but this approach will definetly be broken when we ship "real" private fields in further major release.  
```js
const queue = new FixedQueue()

let buffer = queue.tail
let count = 0

while (buffer !== null) {
  count++
  buffer = buffer.next
}

console.log(`The queue has ${count} buffers`)
```

Reply:
Overall, I'd not recommend to alter the internals just for testing; I've always advocated for testing the outcome rather than expectations.
If we want to check how many internal circular buffers has been made, we can always check the overall outcome, we know that if two circular buffers are made, at least we should have 2048 + 1, this at the `FixedTaskQueue` level.

At the `FixedQueue` we can go deep but always using the interfaces as a bridge and not going beyond

Reply:
I completely agree. Is there anything else blocking the PR from being landed?